caddy/caddymain/run.go

```diff
@@ -127,6 +130,34 @@ func Run() {
 		fmt.Println(caddy.DescribePlugins())
 		os.Exit(0)
 	}
+	if fromJSON {
+		jsonBytes, err1 := ioutil.ReadAll(os.Stdin)
+		if err1 != nil {
+			fmt.Fprintf(os.Stderr, "Read stdin failed: %v", err1)
+			os.Exit(1)
+		}
+		caddyfileBytes, err2 := caddyfile.FromJSON(jsonBytes)
+		if err2 != nil {
+			fmt.Fprintf(os.Stderr, "Convert from JSON failed: %v", err2)
+			os.Exit(2)
+		}
+		fmt.Println(string(caddyfileBytes))
+		os.Exit(0)
+	}
+	if toJSON {
+		caddyfileBytes, err1 := ioutil.ReadAll(os.Stdin)
+		if err1 != nil {
+			fmt.Fprintf(os.Stderr, "Read stdin failed: %v", err1)
+			os.Exit(1)
+		}
+		jsonBytes, err2 := caddyfile.ToJSON(caddyfileBytes)
+		if err2 != nil {
+			fmt.Fprintf(os.Stderr, "Convert to JSON failed: %v", err2)
+			os.Exit(2)
+		}
+		fmt.Println(string(jsonBytes))
+		os.Exit(0)
+	}
```

Comment:
This function is getting long and complex. We should move the handling of fromJSON and toJSON into a separate function.